community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -389,7 +390,19 @@ public void force() throws IOException
         synchronized ( logFile )
         {
             writer.emptyBufferIntoChannelAndClearIt();
+        }
+        // Force the writer outside of the lock.
+        // This allows multiple threads forcing at the same time to piggy-back onto one another.
```

Comment:
@MishaDemianenko: yes that path also exists, but in that path there is already a wider scoped synchronization on the `logFile`.

@martinfurmanski: Interesting. You might be correct about that. Does that affect correctness?